### PR TITLE
plan: stop finding best child task as soon as possible

### PR DESCRIPTION
**Pull request by zz-jason**
Thank you for working on TiDB!

## What have you changed? (mandatory)

If any of the best children tasks is an invalid task, the best task of current operator is also an invalid task. We can stop finding the best tasks of other children operators once an invalid task is returned by a child, which helps us to reduce the overhead of exhausting useless physical plans of child operators and finding the useless best task of each child.

## What are the type of the changes (mandatory)?

Please check the box (by putting an `x` inside the brackets) for this PR:

- Improvement 

## How has this PR been tested (mandatory)?
- unit test
- explain test

## Does this PR affect documentation (docs/docs-cn) update? (optional)

No

## Add a few positive/negative examples (optional)

Please read TiDB's [CONTRIBUTING](https://github.com/pingcap/tidb/blob/master/CONTRIBUTING.md) document **before** filing this PR.

### CODE REVIEW

**Comment by shenli** [comment]
Will this PR improvement the performance of optimizer in OLTP scenario?

**Comment by zz-jason** [comment]
@shenli Not very much, it may be helpful when we have joins.

File: plan/find_best_task.go [hunk 1]
```diff
@@ -92,16 +92,23 @@ func (p *baseLogicalPlan) findBestTask(prop *requiredProp) (bestTask task, err e
...
 			if err != nil {
 				return nil, errors.Trace(err)
 			}
+			if childTask != nil && childTask.invalid() {
+				break
+			}
 			childTasks = append(childTasks, childTask)
 		}
 
+		if len(childTasks) != len(p.children) {
```
[the remark this answers is not in the document]
Comment (zz-jason) [reply plan/find_best_task.go]: This check makes sure that there is no invalid child task. See line 102 for more detail.

[the remark this answers is not in the document]
Comment (shenli) [reply plan/find_best_task.go]: Please add a comment here to describe why to continue.

**Comment by shenli** [comment]
LGTM

**Pull request merged**